frontend/src/components/artifact/ArtifactTable.js

```diff
@@ -1,37 +1,75 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow } from '@material-ui/core';
+import { Table, TableBody, TableCell, TableContainer, TableHead, TableRow, TableSortLabel } from '@material-ui/core';
 
 import ArtifactTableRow from './ArtifactTableRow';
-import { sortMostRecent } from 'utils/sort';
+import { sortByName, sortByVersion, sortByDateEdited, sortByDateCreated } from 'utils/artifactSort';
 import artifactProps from 'prop-types/artifact';
 
-const ArtifactTable = ({ artifacts, handleDeleteArtifact, handleUpdateArtifact }) => (
-  <TableContainer>
-    <Table aria-label="artifact table">
-      <TableHead>
-        <TableRow>
-          <TableCell>Artifact Name</TableCell>
-          <TableCell>Version</TableCell>
-          <TableCell>Last Updated</TableCell>
-          <TableCell></TableCell>
-        </TableRow>
-      </TableHead>
-
-      <TableBody>
-        {artifacts.sort(sortMostRecent).map(artifact => (
-          <TableRow key={artifact._id}>
-            <ArtifactTableRow
-              artifact={artifact}
-              handleDeleteArtifact={handleDeleteArtifact}
-              handleUpdateArtifact={handleUpdateArtifact}
-            />
+const ArtifactTable = ({ artifacts, handleDeleteArtifact, handleUpdateArtifact }) => {
+  const [selectedColumn, setSelectedColumn] = useState(2);
+  const [columnOrder, setColumnOrder] = useState(true);
+
+  const columns = [
+    { columnName: 'Artifact Name', columnSortHandler: sortByName },
+    { columnName: 'Version', columnSortHandler: sortByVersion },
+    { columnName: 'Last Changed', columnSortHandler: sortByDateEdited },
+    { columnName: 'Date Created', columnSortHandler: sortByDateCreated }
+  ];
+
+  const handleLabelClick = id => {
```

Comment:
Can we change the name of this function to be `handleRequestSort` to be more explicit?

Reply:
resolved